processor/prio_processor/indexing.py

```diff
@@ -0,0 +1,123 @@
+"""Map Prio-aggregated data to their corresponding origins."""
+import json
+
+import click
+from jsonschema import validate
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import explode, udf
+from pyspark.sql.types import (
+    ArrayType,
+    IntegerType,
+    StringType,
+    StructField,
+    StructType,
+)
+
+
+def validate_origins(origins):
+    schema = {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "name": {"type": "string"},
+                "hash": {"type": "string"},
+                "index": {"type": "integer", "minimum": 0},
+            },
+        },
+    }
+    validate(instance=origins, schema=schema)
+
+
+def extract(spark, input):
+    return spark.read.json(input)
+
+
+def transform(aggregates, config, origins):
+    @udf(
+        ArrayType(
+            StructType(
+                [
+                    StructField("batch_id", StringType(), False),
+                    StructField("origin", StringType(), False),
+                    StructField("hash", StringType(), False),
+                    StructField("index", IntegerType(), False),
+                    StructField("aggregate", IntegerType(), False),
+                ]
+            )
+        )
+    )
+    def _apply_structure(batch_id, payload):
+        """Create a user-defined function that maps partitioned batch-ids into
+        list of structures containing the aggregate value and its metadata."""
+
+        # assumption: hyphens are used to define a partition of origins
+        if batch_id not in config:
+            return []
+
+        # currently all batch-ids contain a single hyphen with 2 parts
+        split = batch_id.split("-")
```

Comment:
Should we assert on this condition?

Reply:
Sure, it seems reasonable -- otherwise it should raise an IndexError on line 61.